test/foundry/deploy/11_HolographFactory.t.sol

```diff
@@ -0,0 +1,353 @@
+
+    vm.expectRevert(bytes(ErrorConstants.INVALID_SIGNATURE_ERROR_MSG));
+    vm.prank(deployer);
+    holographFactory.deployHolographableContract(deployConfig, signature, owner);
+  }
+
+  /**
+   * @notice Test the deployHolographableContract function Revert if the contract was already deployed
+   * @dev  Refers to the hardhat test with the description 'should fail contract was already deployed'
+   */
+  function testDeployRevertContractAlreadyDeployed() public {
+    //TODO the internal function _isContract is not working.
+    vm.skip(true);
+    (DeploymentConfig memory deployConfig, bytes32 hashHtokenEth) = getConfigHtokenETH();
+    (uint8 v, bytes32 r, bytes32 s) = vm.sign(
+      privateKeyDeployer,
+      HelperSignEthMessage.toEthSignedMessageHash(hashHtokenEth)
+    );
+    console.logBytes32(hashHtokenEth);
+    Verification memory signature = Verification({v: v, r: r, s: s});
+
+    bytes32 hash = keccak256(
+      abi.encodePacked(
+        deployConfig.contractType,
+        deployConfig.chainType,
+        deployConfig.salt,
+        keccak256(deployConfig.byteCode),
+        keccak256(deployConfig.initCode),
+        deployer
+      )
+    );
+    console.logBytes32(hash);
+    console.logBytes32(hashHtokenEth);
+    bytes memory holographerBytecode = type(Holographer).creationCode;
+    // console.logBytes32(bytes32(holographerBytecode));
+    address holographerAddress = address(
+      uint160(
+        uint256(
+          keccak256(abi.encodePacked(bytes1(0xff), address(holographFactory), hash, keccak256(holographerBytecode)))
+        )
+      )
+    );
+    bool isContract = _isContract(holographerAddress);
+    console.log("holographerAddress", holographerAddress);
+    console.log(isContract);
+
+    vm.expectRevert(bytes(ErrorConstants.ALREADY_DEPLOYED_ERROR_MSG));
+    // vm.prank(deployer);
+    holographFactory.deployHolographableContract(deployConfig, signature, deployer);
+  }
+  //TODO remove this function, its olny for test the previous test
+  function _isContract(address contractAddress) private view returns (bool) {
+    bytes32 codehash;
+    assembly {
+      codehash := extcodehash(contractAddress)
+    }
+    console.logBytes32(codehash);
+    return (codehash != 0x0 && codehash != 0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470);
+  }
+
+  /**
+   * @notice Test the deployHolographableContract function Revert if the signature R is invalid
+   * @dev  Refers to the hardhat test with the description 'should fail with invalid signature if signature.r is incorrect'
+   */
+  function testDeployRevertSignatureRIncorrect() public {
+    (DeploymentConfig memory deployConfig, bytes32 hashHtokenEth) = getConfigHtokenETH();
+    (uint8 v, bytes32 r, bytes32 s) = vm.sign(
+      privateKeyDeployer,
+      HelperSignEthMessage.toEthSignedMessageHash(hashHtokenEth)
+    );
+    Verification memory signature = Verification({v: v, r: bytes32(invalidSignature), s: s});
```

Comment:
@mmvazzano this should be using a valid signature.

Reply:
that's old and it's from the other test. She need to get out of that branch because she needed the functionalities in common.
That's why I'm asking to catch up with develop
Now i will notify she to update the branch with develop

Reply:
Done!